Sources/AWSLambdaRuntimeCore/DetachedTasks.swift

```diff
@@ -0,0 +1,169 @@
+    ///
+    /// - Warning: This method does not actually stop the execution of the
+    ///   detached task, it only prevents the runtime from waiting for it before
+    ///   `/next` is invoked.
+    ///
+    /// - Parameter key: The `RegistrationKey` of the task to cancel.
+    public func unsafeCancel(_ key: RegistrationKey) {
+        // To discuss:
+        // Canceling the execution doesn't seem to be an easy
+        // task https://github.com/apple/swift-nio/issues/2087
+        //
+        // While removing the handler will allow the runtime
+        // to invoke `/next` without actually awaiting for the
+        // task to complete, it does not actually cancel
+        // the execution of the dispatched task.
+        // Since this is a bit counter-intuitive, we might not
+        // want this method to exist at all.
+        self.storage.remove(key)
+    }
+
+    /// Awaits all registered tasks to complete.
+    ///
+    /// - Returns: An `EventLoopFuture<Void>` that completes when all tasks have finished.
+    internal func awaitAll() -> EventLoopFuture<Void> {
+        let tasks = storage.tasks
+        if tasks.isEmpty {
+            return context.eventLoop.makeSucceededVoidFuture()
+        } else {
+            return EventLoopFuture.andAllComplete(tasks.map(\.value.task), on: context.eventLoop).flatMap {
+                self.awaitAll()
+            }
+        }
+    }
+}
+
+extension DetachedTasksContainer {
+    /// Lambda detached task registration key.
+    public struct RegistrationKey: Hashable, CustomStringConvertible {
+        var value: String
+
+        init() {
+            // UUID basically
+            self.value = UUID().uuidString
+        }
+
+        public var description: String {
+            self.value
+        }
+    }
+}
+
+extension DetachedTasksContainer {
+    fileprivate final class Storage {
+        private let lock: NIOLock
+        
+        private var map: [RegistrationKey: (name: String, task: EventLoopFuture<Void>)]
+
+        init() {
+            self.lock = .init()
+            self.map = [:]
+        }
+
+        func add(key: RegistrationKey, name: String, task: EventLoopFuture<Void>) {
+            self.lock.withLock {
+                self.map[key] = (name: name, task: task)
+            }
+        }
+
+        func remove(_ key: RegistrationKey) {
+            self.lock.withLock {
+                self.map[key] = nil
+            }
+        }
+
+        var tasks: [RegistrationKey: (name: String, task: EventLoopFuture<Void>)] {
+            self.lock.withLock {
+                self.map
+            }
+        }
+    }
+}
+
+// Ideally this would not be @unchecked Sendable, but Sendable checks do not understand locks
+// We can transition this to an actor once we drop support for older Swift versions
+extension DetachedTasksContainer: @unchecked Sendable {}
+extension DetachedTasksContainer.Storage: @unchecked Sendable {}
+extension DetachedTasksContainer.RegistrationKey: Sendable {}
```

Comment:
We must make sure that all those types are `Sendable` without using `@unchecked`. I'm sure we can achieve this.

Reply:
I believe we can, I took inspiration from [Terminator.swift](https://github.com/swift-server/swift-aws-lambda-runtime/blob/872183bbe99aaa26e52f9d4baef6adaf815f45df/Sources/AWSLambdaRuntimeCore/Terminator.swift#L143) for that part, but I'll change it.